OmegaExamples/src/probmods/generative-models.jl

```diff
@@ -0,0 +1,612 @@
+Programming languages are formal systems for describing what (deterministic) computation a computer should do. Modern programming languages offer a wide variety of different ways to describe computation; each makes some processes simple to describe and others more complex. However, a key tenet of computer science is that all of these languages have the same fundamental power: any computation that can be described with one programming language can be described by another. (More technically this Church-Turing thesis posits that many specific computational systems capture the set of all effectively computable procedures. These are called universal systems.)"
+
+# ╔═╡ cb28a5c4-a721-4a53-a179-adedf4f71237
+md"""
+[Omega](https://github.com/zenna/Omega.jl) is a programming language that describes probabilistic computation. 
+The key idea is that in Omega you can express stochastic processes and not just deterministic functions (like _and_).
+"""
+
+# ╔═╡ 0ce18f90-f2ef-490a-a7c4-40a6ccfa214a
+md"""
+A basic object in Omega is a random variable class, from which you can generate random variables. For instance, a random variable that is of Bernoulli distribution is generated by -
+"""
+
+# ╔═╡ 354e3b5a-4d1e-43ee-b1bb-e8620fba49e7
+a = 0 ~ Bernoulli()
+
+# ╔═╡ 4c915901-e987-4552-b5f7-bc25ee6edeec
+md"Here, `Bernoulli()` is a random class and to get a member (a random variable, which can be sampled from) of the class we use `~`. So, `0 ~ Bernoulli()` is the `0`th member of Bernoulli class."
+
+# ╔═╡ 7d3c8448-090d-46b4-a6b6-ba20ff617be8
+md"A class is analogous to a [*plate*](https://en.wikipedia.org/wiki/Plate_notation) in Bayesian networks."
+
+# ╔═╡ 9dca3d31-8ace-4c64-996c-d9d9e87e6ee4
+md"### Sampling "
+
+# ╔═╡ cc0c0d70-5197-4b2f-b32d-6ac52b8daf27
+md"`randsample` is used to sample from a random variable:"
+
+# ╔═╡ d607d2e0-84a5-4dfd-a281-f5c64cfe1e2d
+randsample(a)
+
+# ╔═╡ 5a10bc0d-d676-4941-a1d5-41790ef255ec
+md"Run this program a few times. You will get back a different sample on each execution."
+
+# ╔═╡ 02ca238e-5a17-42dc-92a1-be975db0e7cd
+md"To understand how `randsample` works, a key concept to know is the Ω object in Omega.jl."
+
+# ╔═╡ 4ea27371-2e73-4a69-887c-b9f519fbf583
+md"""
+There are a few perspectives on what Ω is:
+
+1. The sample space in probability theory (often denoted $\Omega$)
+2. Space of exogenous variables in as causal graphical modelling nomenclature (often denoted $U$)
+3. The random number generator `rng::Random.AbstractRNG`.  Implementation wise, Omega.jl tries to be compatible with any model written as a function `f(rng::AbstractRNG)`
+"""
+
+# ╔═╡ d70b97b5-af55-46f9-904e-309713603155
+md"""
+Conceptually, sampling from a random variable is done in the following two distinct steps:
+1. Construct a random $ω : Ω$, where $Ω$ can be though of as the space of exogenous variables as in causal graphical modelling nomenclature or simply the random number generator, `rng::Random.AbstractRNG`.
```

Comment:
- What does "Construct a random ω : Ω" mean?
- Why are you using mathtype with Greek letters?  Why no write either "Construct a random \$\omega : \Omega$\" or "Construct a random ω : Ω" without the dollar signs?  Perhaps you want the mathtype formatting?

Reply:
A note on format: why are you using mathtype with Greek letters?  Why no write either "Construct a random \$\omega : \Omega$\" or "Construct a random ω : Ω" without the dollar signs?  Perhaps you want the mathtype formatting?

Reply:
I'm not sure it's necessary for the user in the first place since it describes internals of Omega, it's not something the user needs to do, `randsample` does it. Should I remove this portion? 
I included it because we write functions of `\omega::\Omega`  to construct random variables (which `randsample` can be used on) and I wasn't sure how to introduce that.